### PR TITLE
feature: plugin `limit-count` support to use `redis` cluster

### PR DESCRIPTION
**Pull request by membphis**
### What this PR does / why we need it:

Support for using redis cluster in limit-count plugin.

fix https://github.com/apache/apisix/issues/638

continue PR: https://github.com/apache/apisix/pull/2340

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

File: apisix/plugins/limit-count/limit-count-redis-cluster.lua [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        if not lock then
+            return false, "failed to create lock: " .. err
+        end
+
+        local elapsed, err = lock:lock(key)
+        if not elapsed then
+            return false, "failed to acquire the lock: " .. err
+        end
+
+        ret = red:ttl(key)
```
Comment (nic-chen) [inline apisix/plugins/limit-count/limit-count-redis-cluster.lua:100]: why read ttl twice ? a little confuse

**Review by gxthrj — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

File: t/plugin/limit-count-redis-cluster.t [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+--- pipelined_requests eval
+["GET /hello1", "GET /hello", "GET /hello2", "GET /hello", "GET /hello"]
+--- error_code eval
+[404, 503, 404, 503, 503]
+--- no_error_log
+[error]
+
+
+
+=== TEST 6: up the limit
```
Comment (moonming) [inline t/plugin/limit-count-redis-cluster.t:214]: Repeated test case

**Review by moonming — CHANGES_REQUESTED** [review]

File: apisix/plugins/limit-count/limit-count-redis-cluster.lua (hunk 1 above)
Comment (moonming) [reply apisix/plugins/limit-count/limit-count-redis-cluster.lua:100]: why？

File: apisix/plugins/limit-count/limit-count-redis-cluster.lua [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+    local self = {
+        limit = limit, window = window, conf = conf,
+        plugin_name = plugin_name, red_cli =red_cli
+    }
+
+    return setmetatable(self, mt)
+end
+
+
```
Comment (moonming) [inline apisix/plugins/limit-count/limit-count-redis-cluster.lua:75]: I don't think copy-paste is good

File: apisix/plugins/limit-count/limit-count-redis-cluster.lua [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            if not ok then
+                return false, "failed to unlock: " .. err
+            end
+
+            ret, err = red:set(key, limit -1, "EX", window)
+            if not ret then
+                return nil, err
+            end
+
+            return 0, limit -1
```
Comment (moonming) [inline apisix/plugins/limit-count/limit-count-redis-cluster.lua:112]: Code style

File: t/plugin/limit-count-redis-cluster.t [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "uri": "/hello",
+                    "plugins": {
+                        "limit-count": {
+                            "count": 2,
```
Comment (moonming) [inline t/plugin/limit-count-redis-cluster.t:130]: There is no test case to cover what happens after the configuration value is changed

**Review by moonming — CHANGES_REQUESTED** [review]

File: apisix/plugins/limit-count/limit-count-redis-cluster.lua (hunk 3 above)
Comment (membphis) [reply apisix/plugins/limit-count/limit-count-redis-cluster.lua:75]: @moonming I think we can merge this PR now. please take a look at this PR.

**Review by moonming — APPROVED** [review]

**Pull request merged**